streamalert/shared/config.py

```diff
@@ -443,3 +443,60 @@ def _validate_sources(cluster_name, data_sources, existing_sources):
             existing_sources.add(source)
 
 # FIXME (derek.wang) write a configuration validator for lookuptables (new one)
+
+def artifact_extractor_enabled(config, log_name=None):
```

Comment:
I recommend two functions:

```
def artifact_extractor_enabled_for_log(config, log_name):
  return _artifact_extractor_enabled_helper(config, log_name=log_name)

def artifact_extractor_enabled(config):
  return _artifact_extractor_enabled_helper(config, log_name=None)
```

I'm not a fan of optional arguments that alter the control flow of functions. This is acceptable for private methods used to DRY out blocks of code but it's better to be explicit (through the function public interface) exactly what a function is intended for

Reply:
🐟 Okay, this is your 2nd request of changes when I use a optional arguments. I will address your concern this time.